### DISCUSSION TITLE
Copy Value returns entire row

### DISCUSSION THREAD

**Discussion by mohamed-umar** [nick-keller/react-datasheet-grid#116] [Q&A] [answered]
Hey !!! 
copyValue() function Returns entire row even one column of row is selected. Can you please return Only selected values instead of entire rows.

Calling the CopyValue() out of loop with CopyData should return the ONLY SELECTED value for ONE TIME.
Can you please make the change if possible..

**Comment by nick-keller** [comment]
Hello @mohamed-umar ,
I am not sure I understand what is going on, could you please provide so code example?

**Reply by mohamed-umar** [reply]
<img width="1543" alt="Screenshot 2022-01-28 at 2 46 55 PM" src="https://user-images.githubusercontent.com/59644436/151520039-2cc1d2f6-4b40-4f33-929e-de4386fc67cf.png">

**Reply by mohamed-umar** [reply]
Here I selected first 2 columns of row 1, but it returns entire row instead of selected value..
@nick-keller

**Reply by mohamed-umar** [reply]
The issue am facing is..
Am currently using webpack@4.41.2, in which webpack-dev-server uses 'html-entites' of older version.
But in your code Datasheetgrid.jsx: line: 614 you are using html-entites' of latest version.
But updating 'html-entites' breaks my webpack..


So i tried to use copyValue function , but it returned entire row insated of selected value...so am finding it difficult to process

**Reply by nick-keller** [reply]
I have no idea what `valueCopy` is, I cannot find it anywhere in the codebase, you need to provide some code if you'd like to be helped.

**Comment by mohamed-umar** [comment]
@nick-keller Here is the function mentioned in Doc..
<img width="1543" alt="Screenshot 2022-01-28 at 11 03 43 PM" src="https://user-images.githubusercontent.com/59644436/151594453-bbfc5cf4-9fad-4a22-9279-1705386be8b5.png">

In Code  path: components/DataSheetGrid.jsx line 595
<img width="1543" alt="Screenshot 2022-01-28 at 11 05 32 PM" src="https://user-images.githubusercontent.com/59644436/151594697-69c5a4d0-e96b-43c2-bc6c-a26a8d14cab8.png">

**Reply by mohamed-umar** [reply]
@nick-keller In the copy event , your are calling the copyValue() inside the loop , also instead of selected value whole row data is passed. So its impossible to map the selected value .

**Reply by mohamed-umar** [reply]
@nick-keller Any updates ??

**Reply by nick-keller** [reply]
Hey @mohamed-umar ,
This is not a bug, according to the doc this is exactly how `copyValue` should work so I will not change it.

I believe that you are trying to use it to do something else because you have a dependencies issue and the built-in copy does not work in your setup. Apparently, you are not willing to share your code, so I still have no idea what `valueCopy` is or what exactly you are doing.

I might try to remove the dependency on html-entities and build it in-house, that might solve your issue.

**Reply by mohamed-umar** [reply]
Yeah, that too should help me..
thanks @nick-keller

**Reply by mohamed-umar** [reply]
Sorry to ask, When can i expect the changes to be published @nick-keller

**Reply by nick-keller** [reply]
No idea, but you can make it happen really quick if you want to submit a PR ;)

**Reply by mohamed-umar** [reply]
Hey @nick-keller !!! I will do the changes and raise PR..
But cannot able to create a branch from master branch..can you help me on this..

**Comment by nick-keller** [comment, marked as answer]
I just published a new version that does not require html-entities to work, that should fix your problem ;)
